base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,310 @@
+                    continue
+
+                if s.id == 'KRA':
+                    if h.Clone == 'TRUE':
+                        self.clone_kras.append(h)
+                    else:
+                        self.master_kras.append(h)
+                    continue
+
+                if s.id == 'OCSP':
+                    if h.Clone == 'TRUE':
+                        self.clone_ocsps.append(h)
+                    else:
+                        self.master_ocsps.append(h)
+                    continue
+
+                if s.id == 'TPS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tpss.append(h)
+                    else:
+                        self.master_tpss.append(h)
+                    continue
+
+                if s.id == 'TKS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tkss.append(h)
+                    else:
+                        self.master_tkss.append(h)
+                    continue
+
+        return domain_data
+
+    def get_security_domain_ca(self):
+        sec_domain = None
+        sechost = None
+        secport = None
+        for subsystem in self.instance.get_subsystems():
+            if(subsystem.type == 'CA'):
+                # make sure this CA is the security domain
+                service_host = subsystem.config.get('service.machineName')
+                service_port = subsystem.config.get('service.unsecurePort')
+                sechost = subsystem.config.get('securitydomain.host')
+                secport = subsystem.config.get('securitydomain.httpport')
+
+                if sechost == service_host and secport == service_port:
+                    sec_domain = subsystem
+                    break
+
+        if sec_domain is not None:
+            self.security_domain = sec_domain
+            # Set some vars we will be using later
+            self.db_dir = self.security_domain.config.get('jss.configDir')
+            self.subsystem_token = self.security_domain.config.get('ca.subsystem.tokenname')
+            self.passwd = self.instance.get_password(self.subsystem_token)
+
+        return sec_domain, sechost, secport
+
+    def check_ca_clones(self):
+        for host in self.clone_cas:
+            cur_clone_msg = ' Host: ' + host.Hostname + ' Port: ' + host.SecurePort
+            # Reach out and get some certs, to serve as a data and connectivity check
+            try:
+                connection = PKIConnection(protocol='https',
+                                           hostname=host.Hostname,
+                                           port=host.SecurePort,
+                                           verify=False)
+
+                cert_client = CertClient(connection)
+                # get the first 5 in case we cant to make a sanity check of replicated data
+                certs = cert_client.list_certs(size=5)
+
+                if certs is not None and len(certs.cert_data_info_list) == 5:
```

Comment:
Why are we using `5` as a hard limit to pass our test? What happens in a freshly installed sub-CA and clone sub-CA ? IIUC, there will be only 4 certs. no?

Reply:
I just tried to pick some arbitrary number of certs that would tell us the thing has data. So no problem to change it to something smaller.

Reply:
Fixed, made it 3 just to be safe.